src/Mvc/Mvc.Core/src/ModelBinding/Metadata/SystemTextJsonValidationMetadataProvider.cs

```diff
@@ -55,7 +55,7 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
 
         var propertyName = ReadPropertyNameFrom(context.Attributes);
 
-        if (string.IsNullOrEmpty(propertyName))
+        if (!string.IsNullOrEmpty(propertyName))
```

Comment:
Good call, I misinterpreted the code as running the naming policy against `propertyName`. In that case I'll close the PR and transfer https://github.com/dotnet/runtime/issues/84830 to aspnetcore so that you folks can make the appropriate fix.

Reply:
I ended up updating the PR so that it includes a null check for `context.Key.Name`. That should emulate the semantics of `JsonNamingPolicy.CamelCase` for other naming policies.